tests/boards/nrf/coresight_stm/pytest/test_stm.py

```diff
@@ -0,0 +1,387 @@
+    # check value
+    latency_msg_1 = float(latency_msg_1_str)
+    threshold = (constraints.log_1_arg) * (1 + constraints.tolerance)
+    assert (
+        latency_msg_1 < threshold
+    ), f"{core}: Timing for log message with 1 argument {latency_msg_1} us exceeds {threshold} us"
+
+    # 2 arguments
+    latency_msg_2_str = re.search(
+        rf"{core}: Timing for log message with 2 arguments: (.+)us", output
+    ).group(1)
+    assert latency_msg_2_str is not None, "Timing for log message with 2 arguments NOT found"
+    # check value
+    latency_msg_2 = float(latency_msg_2_str)
+    threshold = (constraints.log_2_arg) * (1 + constraints.tolerance)
+    assert (
+        latency_msg_2 < threshold
+    ), f"{core}: Timing for log message with 2 arguments {latency_msg_2} us exceeds {threshold} us"
+
+    # 3 arguments
+    latency_msg_3_str = re.search(
+        rf"{core}: Timing for log message with 3 arguments: (.+)us", output
+    ).group(1)
+    assert latency_msg_3_str is not None, "Timing for log message with 3 arguments NOT found"
+    # check value
+    latency_msg_3 = float(latency_msg_3_str)
+    threshold = (constraints.log_3_arg) * (1 + constraints.tolerance)
+    assert (
+        latency_msg_3 < threshold
+    ), f"{core}: Timing for log message with 3 arguments {latency_msg_3} us exceeds {threshold} us"
+
+    # string argument
+    latency_msg_string_str = re.search(
+        rf"{core}: Timing for log_message with string: (.+)us", output
+    ).group(1)
+    assert latency_msg_string_str is not None, "Timing for log_message with string NOT found"
+    # check value
+    latency_msg_string = float(latency_msg_string_str)
+    threshold = (constraints.log_str) * (1 + constraints.tolerance)
+    assert (
+        latency_msg_string < threshold
+    ), f"{core}: Timing for log message with string {latency_msg_string} us exceeds {threshold} us"
+
+    # tracepoint
+    latency_tracepoint_str = re.search(rf"{core}: Timing for tracepoint: (.+)us", output).group(1)
+    assert latency_tracepoint_str is not None, "Timing for tracepoint NOT found"
+    # check value
+    latency_tracepoint = float(latency_tracepoint_str)
+    threshold = (constraints.tracepoint) * (1 + constraints.tolerance)
+    assert (
+        latency_tracepoint < threshold
+    ), f"{core}: Timing for tracepoint {latency_tracepoint} us exceeds {threshold} us"
+
+    # tracepoint with 32-bit data
+    latency_tracepoint_d32_str = re.search(
+        rf"{core}: Timing for tracepoint_d32: (.+)us", output
+    ).group(1)
+    assert latency_tracepoint_d32_str is not None, "Timing for tracepoint_d32 NOT found"
+    # check value
+    latency_tracepoint_d32 = float(latency_tracepoint_d32_str)
+    threshold = (constraints.tracepoint_d32) * (1 + constraints.tolerance)
+    assert (
+        latency_tracepoint_d32 < threshold
+    ), f"{core}: Timing for tracepoint_d32 {latency_tracepoint_d32} us exceeds {threshold} us"
```

Comment:
Why not use arrays and loop?

Reply:
Each Loop iteration requires different regex, threshold to compare, two error message strings.
This runs on PC so code optimization isn't top priority.

Reply:
Actually, loop with a dictionary looks simpler.